pkg/services/alerting/notifiers/victorops.go

```diff
@@ -0,0 +1,87 @@
+package notifiers
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/grafana/grafana/pkg/bus"
+	"github.com/grafana/grafana/pkg/log"
+	"github.com/grafana/grafana/pkg/metrics"
+	m "github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/alerting"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+const AlertStateCritical = "CRITICAL"
+
+func init() {
+	alerting.RegisterNotifier("victorops", NewVictoropsNotifier)
+}
+
+func NewVictoropsNotifier(model *m.AlertNotification) (alerting.Notifier, error) {
+	url := model.Settings.Get("url").MustString()
+	if url == "" {
+		return nil, alerting.ValidationError{Reason: "Could not find url property in settings"}
+	}
+
+	return &VictoropsNotifier{
+		NotifierBase: NewNotifierBase(model.Id, model.IsDefault, model.Name, model.Type, model.Settings),
+		Url:          url,
+		log:          log.New("alerting.notifier.victorops"),
+	}, nil
+}
+
+type VictoropsNotifier struct {
+	NotifierBase
+	Url string
+	log log.Logger
+}
+
+func (this *VictoropsNotifier) Notify(evalContext *alerting.EvalContext) error {
+	this.log.Info("Executing victorops notification", "ruleId", evalContext.Rule.Id, "notification", this.Name)
+	metrics.M_Alerting_Notification_Sent_Victorops.Inc(1)
+
+	fields := make([]map[string]interface{}, 0)
+	fieldLimitCount := 4
+	for index, evt := range evalContext.EvalMatches {
+		fields = append(fields, map[string]interface{}{
+			"title": evt.Metric,
+			"value": evt.Value,
+			"short": true,
+		})
+		if index > fieldLimitCount {
+			break
+		}
+	}
+
+	if evalContext.Error != nil {
+		fields = append(fields, map[string]interface{}{
+			"title": "Error message",
+			"value": evalContext.Error.Error(),
+			"short": false,
+		})
+	}
+
+  messageType := evalContext.Rule.State
+  if evalContext.Rule.State == m.AlertStateAlerting { // translate 'Alerting' to 'CRITICAL' (Victorops analog)
+    messageType = AlertStateCritical
+  }
+
+	body := map[string]interface{}{
+    "message_type": messageType,
+    "entity_id": evalContext.Rule.Name,
+    "timestamp": time.Now().Unix(),
```

Comment:
Does VO use nanoseconds for it's timestamps?  That's surprising to me.  If it's epoch milliseconds instead, you can use:

```
time.Now().Unix() / 1e6
```